pkg/pipeline/transform/transform_generic.go

```diff
@@ -24,16 +24,22 @@ import (
 )
 
 type Generic struct {
-	Rules []api.GenericTransformRule
+	policy string
+	rules  []api.GenericTransformRule
 }
 
 // Transform transforms a flow to a new set of keys
 func (g *Generic) Transform(input []config.GenericMap) []config.GenericMap {
-	log.Debugf("f = %v", g)
+	log.Debugf("entering Generic Transform g = %v", g)
 	output := make([]config.GenericMap, 0)
 	for _, entry := range input {
-		outputEntry := make(config.GenericMap)
-		for _, transformRule := range g.Rules {
+		outputEntry := entry
+		if g.policy == "replace_keys" {
```

Comment:
Do we want to panic on an invalid policy value? I mean, if the user writes `replace-keys` with a dash instead of an underline, we will treat it as `preserve_original_keys`

Reply:
added panic for illegal policy